ui/panels-plugin/src/plugins/line-chart/LineChart.tsx

```diff
@@ -98,12 +139,69 @@ function LineChart(props: LineChartProps) {
 
   // Sync options with chart instance
   useLayoutEffect(() => {
-    // Can't set options if no chart yet
     if (chart === undefined) return;
 
+    if (option.series === undefined) {
+      chart.showLoading();
+    } else {
+      chart.hideLoading();
+    }
+
     chart.setOption(option);
   }, [chart, option]);
 
+  // Populate tooltip data from getZr cursor coordinates
+  useEffect(() => {
+    if (chart === undefined || option.series === undefined) return;
+
+    const chartWidth = chart.getWidth();
+    const xAxisInterval = timeScale ? timeScale.stepMs : 0;
+    const xBuffer = xAxisInterval * 0.5;
+
+    const yAxisInterval = chart['_model'].getComponent('yAxis').axis.scale._interval;
+    const yBuffer = yAxisInterval * 0.5;
+
+    let lastPosX = -1;
+    let lastPosY = -1;
+    chart.getZr().on('mousemove', (params) => {
```

Comment:
Is there some sort of "clean up" or "unsubscribe" we need to do for this listener for when the component is unmounted or the dependencies change? Typically that'd go in the `useEffect` disposer...

Reply:
Fixed in [this commit](https://github.com/perses/perses/pull/266/commits/67b7fc11e8ea2f2c4c5a0218028c275ea3e7bbc7)